flake.nix

```diff
@@ -30,6 +30,9 @@
 
       perSystem = { pkgs, self', config, ... }: {
         devShells.default = pkgs.mkShell {
+          shellHook = ''
+            export CARGO_HOME=$PWD/.cargo
```

Comment:
The devShell may not always start from project root. But you can use https://github.com/srid/flake-root to determine project root.

We use this trick in nammayatri for setting `CABAL_DIR`: https://github.com/nammayatri/common/blob/main/nix/haskell/no-global-cache.nix

Reply:
Actually I added this for myself, it's not needed. Will remove this